### PR TITLE
Update README.md with release steps

### LINKED ISSUE #72: Provide release steps for Codeflare Instascale
Repository is missing a documentation providing instructions to release a new version. There is a brief mention of image release in https://github.com/project-codeflare/instascale#image-creation , however it is not clear if this is the only step needed to release a new version or if there are some other steps needed.

Readme file should be adjusted to describe steps taken to release a new Instascale version, possibly mentioning any external dependency.

### PR DESCRIPTION
**Pull request by sutaakar**
Fixes #72

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,3 +52,21 @@ go test -v ./controllers/
...
+2. Update version in [VERSION file](https://github.com/project-codeflare/instascale/blob/main/VERSION) to the new release version. Once VERSION file change is merged then [instascale-release.yml](https://github.com/project-codeflare/instascale/actions/workflows/instascale-release.yml) and [go.yml](https://github.com/project-codeflare/instascale/actions/workflows/go.yml) actions are invoked.
+
+> **Note**
+> The VERSION file is going to be removed as part of Instascale release process automation.
+
+3. Verify that [instascale-release.yml](https://github.com/project-codeflare/instascale/actions/workflows/instascale-release.yml) action passed successfully.
+4. Verify that [go.yml](https://github.com/project-codeflare/instascale/actions/workflows/go.yml) action passed successfully.
+5. Create a new release on [release page](https://github.com/project-codeflare/instascale/releases). Provide a tag with same value as in VERSION file formated as semver (with `v` prefix). Provide proper release title and description.
+
+
```
Comment (anishasthana) [inline README.md:72]:
```suggestion
```

**Review by anishasthana — COMMENTED** [review]
This looks broadly correct to me. @Maxusmusti are there any changes we need here?

File: README.md (hunk 1 above)
Comment (sutaakar) [reply README.md]: Done

File: README.md [hunk 2]
```diff
@@ -52,3 +52,20 @@ go test -v ./controllers/
 
 ```
 
+## Release process
+
+1. Update [go.mod](https://github.com/project-codeflare/instascale/blob/main/go.mod)/[go.sum](https://github.com/project-codeflare/instascale/blob/main/go.sum) with newest dependency version of multi-cluster-app-dispatcher:
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply README.md:57]: Well, it is my assumption :)
We should refer to the same MCAD code as is released in the same release as InstaScale.

**Review by Maxusmusti — COMMENTED** [review]
Looks good! Only thing worth mentioning is that @asm582 also likes to have a "release" branch cut as well, named `release-x.y.z`

**Comment by anishasthana** [comment]
> Looks good! Only thing worth mentioning is that @asm582 also likes to have a "release" branch cut as well, named release-x.y.z

I think this is a good chance for us to revisit that. In my opinion having both tags and branches is redundant. We could always create a branch from the tag if we need to backport a fix.

**Review by anishasthana — APPROVED** [review]

**Pull request merged**